airflow/operators/mysql_to_s3_operator.py

```diff
@@ -0,0 +1,95 @@
+from typing import Optional, Union
+
+from io import StringIO
+
+from airflow.hooks.mysql_hook import MySqlHook
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.utils.decorators import apply_defaults
+
+class MySQLToS3Operator(BaseOperator):
+    """
+    Saves data from an specific MySQL query into a file in S3.
+    
+    :param query: the sql query to be executed.
+    :type query: str
+    :param s3_bucket: bucket where the data will be stored
+    :type s3_bucket: str
+    :param s3_key: desired key for the file. It includes the name of the file.
+        If a csv file is wanted, the param must end with ".csv".
+    :type s3_key: str
+    :param mysql_conn_id: reference to a specific mysql database
+    :type mysql_conn_id: str
+     :param aws_conn_id: reference to a specific S3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+        - ``False``: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - ``path/to/cert/bundle.pem``: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    """
+    
+    @apply_defaults
+    def __init__(
+            self,
+            query: str,
+            s3_bucket: str,
+            s3_key: str,
+            mysql_conn_id: str = 'mysql_default',
+            aws_conn_id: str = 'aws_default',
+            verify: Optional[Union[bool, str]] = None,
+            header: Optional[bool] = False,
+            index: Optional[bool] = False,
+            *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.query = query
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.mysql_conn_id = mysql_conn_id
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.header = header
+        self.index = index
+    
+    def execute(self, context, **kwargs):
+        self.hook = MySqlHook(mysql_conn_id=self.mysql_conn_id)
+        self.s3 = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
+        
+        data_df = self.hook.get_pandas_df(self.query)
+        self.log.info("Data from MySQL obtained")
+        
+        file_obj = StringIO()
+        data_df.to_csv(file_obj, header=self.header, index=self.index)
+        self.log.info("File_obj created")
+        
+        self.s3.load_file_obj(file_obj=file_obj,
+                          key=self.s3_key,
+                          bucket_name=self.s3_bucket,
+                          **kwargs)
+        self.log.info("File saved correctly in {s3_bucket}/{s3_key}"
```

Comment:
I think a verification step should be included if we have the log displaying "File saved correctly". Something like the snippet below would do the trick:

```Python
obj = self.s3.get_key(self.s3_key, self.s3_bucket)

try:
    obj.wait_until_exists()
except WaiterError:
    # either raise AirflowException or log warn/error
```